### ISSUE TITLE
overflow does not behave as expected

### ISSUE THREAD

**Issue by fabianmossberg** [spatie/opening-hours#111]
I create opening hours for MONDAY and TUESDAY that are identical.

`$open_with_overflow` and `$open_without_overflow`

I create `$the_time` that is set to Tuesday the 4th of June 2019.

```PHP
//Tuesday 4th of June 2019, 11.35 am
$the_time = new DateTime('2019-06-04 11:35:00');

$open_with_overflow = OpeningHours::create([
    'overflow' => true,
    'monday'    => ['11:00-18:00'],
    'tuesday'    => ['13:37-15:37'],
]
);

$open_without_overflow = OpeningHours::create([
    'overflow' => false,
    'monday'    => ['11:00-18:00'],
    'tuesday'    => ['13:37-15:37'],
   
]
);

var_dump(
    $open_with_overflow->isOpenAt($the_time),
    $open_without_overflow->isOpenAt($the_time)
);
```

These two isOpen calls should both result in the same anser, but the output from the above is:

```
bool(true)
bool(false)
```

If i change the closing hours on monday to 11.34 they both return false.

It seems like the overflow automatically assumes that the previous days closing time applies to the current date.

This should only be true if the previous days closing time happens BEFORE the current days closing time.

**Comment by kylekatarnls** [comment]
It's indeed a bug, `overflow` should do nothing until a range has an end smaller than its start.

**Comment by kylekatarnls** [comment]
Thanks, patch released 2.2.1.
